src/hackathons-collector/hackathons-collector.js

```diff
@@ -24,31 +25,48 @@ import akai from "./assets/akai-logo.svg";
 export default class HackathonsCollector extends Component {
   constructor() {
     super();
-    this.previousEvents = events
+    this.state = {
+      filterWeekends: false,
+      allEvents: events
+    };
+  }
+  render() {
+    const previousEvents = this.state.allEvents
       .filter(event => event.date <= new Date())
+      .filter(event => {
+        return !this.state.filterWeekends || event.date.getDay() > 5;
```

Comment:
We could extract that condition, that's used in `previousEvents` and `events`, but I'm fine with how it looks now 😉

Reply:
I've tried to figure it out using functional programming paradigm, but naming these structures was very tricky so, for now, let's stay with that solution 😉